### ISSUE TITLE
Benchmark performance gap between packing or not

### ISSUE THREAD

**Issue by lmc8133** [open-thoughts/open-thoughts#30]
Thank you for your open source work!
I found that you have applied neat_packing in your training config, however I have compared the benchmark performance (aime24, math500) between using neat_packing or not. My baseline is Qwen2.5-32B-Instruct, here are my results:
| Method                                     | MATH500 | AIME24  |
| ---------------------------------------- | ----- | ------ |
| Qwen2.5-32B-Instruct (baseline)     | 80.40  | 20.00  |
| random1k_openmathreasoning (no pack)     | 85.6  | 36.67  |
| random1k_openmathreasoning (neat_packing) | 77    | 13.33  |

I have also find similar results in [community](https://github.com/hiyouga/LLaMA-Factory/issues/5426)

Have you ever do some experiments for comparing?

**Comment by EtashGuha** [comment]
Hi! So at a 1k scale, not packing will increase your performance a lot since it greatly increases the number of optimization steps you do. However, if you have more data, packing will make less of a difference.

**Comment by EtashGuha** [comment]
Hm that's interesting. We did ablations of packing and not packing and found at the small scale that packing hurt, but at the higher scales it didn't.

**Comment by lmc8133** [comment]
> Hm that's interesting. We did ablations of packing and not packing and found at the small scale that packing hurt, but at the higher scales it didn't.

Here are my two training configs:
# Pack
    --deepspeed examples/deepspeed/ds_z3_config.json \
    --stage sft \
    --do_train \
    --packing True \
    --neat_packing True \
    --enable_liger_kernel \
    --flash_attn fa2 \
    --model_name_or_path /mnt4/data/open_source/Qwen2.5-32B-Instruct/ \
    --dataset openmathreasoning_r1_passrate_3_cot_random1k \
    --dataset_dir data \
    --template qwen \
    --finetuning_type full \
    --output_dir /mnt/ckpt/qwen25_32b_instruct_openmathreasoning_random1k_packing \
    --overwrite_cache \
    --overwrite_output_dir \
    --cutoff_len 16384 \
    --warmup_ratio 0.0 \
    --save_steps 200 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 16 \
    --gradient_accumulation_steps 16 \
    --ddp_timeout 180000000 \
    --learning_rate 5e-6 \
    --lr_scheduler_type cosine \
    --logging_steps 1 \
    --plot_loss \
    --num_train_epochs 15 \
    --bf16 \
    --seed 17 \
    --report_to "none" \

# No pack
    --deepspeed examples/deepspeed/ds_z3_offload_config.json \
    --stage sft \
    --do_train \
    --flash_attn fa2 \
    --model_name_or_path /mnt4/data/open_source/Qwen2.5-32B-Instruct/ \
    --dataset openmathreasoning_r1_passrate_3_cot_random1k \
    --dataset_dir data \
    --template qwen \
    --finetuning_type full \
    --output_dir /mnt/ckpt/qwen25_32b_instruct_openmathreasoning_random1k \
    --overwrite_cache \
    --overwrite_output_dir \
    --cutoff_len 16384 \
    --warmup_ratio 0.0 \
    --save_steps 200 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 16 \
    --gradient_accumulation_steps 16 \
    --ddp_timeout 180000000 \
    --learning_rate 5e-6 \
    --lr_scheduler_type cosine \
    --logging_steps 1 \
    --plot_loss \
    --num_train_epochs 15 \
    --bf16 \
    --seed 17 \
    --report_to "none" \

**Comment by lmc8133** [comment]
> Hm that's interesting. We did ablations of packing and not packing and found at the small scale that packing hurt, but at the higher scales it didn't.

Could you share your results between these two configs?
